### ISSUE TITLE
DOC: sparse: Several functions are missing the 'Returns' section of their docstring

### ISSUE THREAD

**Issue by WarrenWeckesser** [scipy/scipy#22065]
### Issue with current documentation:

The docstrings of the following functions in `scipy.sparse` are missing the 'Returns' section:

* [`diags`](https://scipy.github.io/devdocs/reference/generated/scipy.sparse.diags.html)
* [`diags_array`](https://scipy.github.io/devdocs/reference/generated/scipy.sparse.diags_array.html)
* [`eye`](https://scipy.github.io/devdocs/reference/generated/scipy.sparse.eye.html)
* [`eye_array`](https://scipy.github.io/devdocs/reference/generated/scipy.sparse.eye_array.html)
* [`identity`](https://scipy.github.io/devdocs/reference/generated/scipy.sparse.identity.html)
* [`save_npz`](https://scipy.github.io/devdocs/reference/generated/scipy.sparse.save_npz.html)
* [`spdiags`](https://scipy.github.io/devdocs/reference/generated/scipy.sparse.spdiags.html)

Nothing urgent here--what each function returns is pretty obvious in all cases--but it would be nice to fix these to conform to the docstring standard.

`save_npz` is pretty clearly a "side effects only" function and returns `None`, so it doesn't necessarily need a 'Returns' section.

**Comment by progressive-galib** [comment]
working on it

**Comment by mdhaber** [comment]
It would be pretty easy to add an automated check for this. There are false positives - that is, functions with no return section which we might not care about - and we can add exceptions for those. But it would catch the addition of new functions without important sections.

Adding this snippet to the `check_rest` function of `refguide_check.py`
```python3
        with warnings.catch_warnings():
            warnings.simplefilter('ignore', UserWarning)
            if inspect.isfunction(obj) and not FunctionDoc(obj)['Returns']:
                result = (full_name, False, f"Docstring has no 'Returns' section")
                results.append(result)
```
produced

<details>

==============
scipy.datasets
==============

scipy.datasets.clear_cache
--------------------------

Docstring has no 'Returns' section

scipy.datasets.download_all
---------------------------

Docstring has no 'Returns' section


=========
scipy.fft
=========

scipy.fft.set_workers
---------------------

Docstring has no 'Returns' section

scipy.fft.set_global_backend
----------------------------

Docstring has no 'Returns' section

scipy.fft.skip_backend
----------------------

Docstring has no 'Returns' section

scipy.fft.set_backend
---------------------

Docstring has no 'Returns' section

scipy.fft.register_backend
--------------------------

Docstring has no 'Returns' section

scipy.fft.get_workers
---------------------

Docstring has no 'Returns' section


=============
scipy.fftpack
=============

scipy.fftpack.itilbert
----------------------

Docstring has no 'Returns' section

scipy.fftpack.sc_diff
---------------------

Docstring has no 'Returns' section

scipy.fftpack.ifftn
-------------------

Docstring has no 'Returns' section

scipy.fftpack.shift
-------------------

Docstring has no 'Returns' section

scipy.fftpack.fft2
------------------

Docstring has no 'Returns' section

scipy.fftpack.ifft2
-------------------

Docstring has no 'Returns' section

scipy.fftpack.ss_diff
---------------------

Docstring has no 'Returns' section

scipy.fftpack.ihilbert
----------------------

Docstring has no 'Returns' section

scipy.fftpack.diff
------------------

Docstring has no 'Returns' section


=================
scipy.interpolate
=================

scipy.interpolate.generate_knots
--------------------------------

Docstring has no 'Returns' section


========
scipy.io
========

scipy.io.savemat
----------------

Docstring has no 'Returns' section


===============
scipy.io.matlab
===============

scipy.io.matlab.savemat
-----------------------

Docstring has no 'Returns' section


================
scipy.io.wavfile
================

scipy.io.wavfile.write
----------------------

Docstring has no 'Returns' section


==========================
scipy.linalg.interpolative
==========================

scipy.linalg.interpolative.rand
-------------------------------

Docstring has no 'Returns' section

scipy.linalg.interpolative.seed
-------------------------------

Docstring has no 'Returns' section


==============
scipy.optimize
==============

scipy.optimize.fixed_point
--------------------------

Docstring has no 'Returns' section

scipy.optimize.linprog_verbose_callback
---------------------------------------

Docstring has no 'Returns' section

============
scipy.signal
============

scipy.signal.cheb1ap
--------------------

Docstring has no 'Returns' section

scipy.signal.buttap
-------------------

Docstring has no 'Returns' section

scipy.signal.ellipap
--------------------

Docstring has no 'Returns' section

scipy.signal.cheb2ap
--------------------

Docstring has no 'Returns' section


============
scipy.sparse
============

scipy.sparse.diags
------------------

Docstring has no 'Returns' section

scipy.sparse.diags_array
------------------------

Docstring has no 'Returns' section

scipy.sparse.save_npz
---------------------

Docstring has no 'Returns' section

scipy.sparse.eye_array
----------------------

Docstring has no 'Returns' section

scipy.sparse.identity
---------------------

Docstring has no 'Returns' section

scipy.sparse.eye
----------------

Docstring has no 'Returns' section

scipy.sparse.spdiags
--------------------

Docstring has no 'Returns' section


===================
scipy.sparse.linalg
===================

scipy.sparse.linalg.aslinearoperator
------------------------------------

Docstring has no 'Returns' section

scipy.sparse.linalg.use_solver
------------------------------

Docstring has no 'Returns' section


=============
scipy.special
=============

scipy.special.pro_cv_seq
------------------------

Docstring has no 'Returns' section

scipy.special.kelvin_zeros
--------------------------

Docstring has no 'Returns' section

scipy.special.lpn
-----------------

Docstring has no 'Returns' section

scipy.special.obl_cv_seq
------------------------

Docstring has no 'Returns' section

scipy.special.lqn
-----------------

Docstring has no 'Returns' section

==================
scipy.stats.mstats
==================

scipy.stats.mstats.trimboth
---------------------------

Docstring has no 'Returns' section

scipy.stats.mstats.trimmed_var
------------------------------

Docstring has no 'Returns' section

scipy.stats.mstats.trima
------------------------

Docstring has no 'Returns' section

scipy.stats.mstats.rsh
----------------------

Docstring has no 'Returns' section

scipy.stats.mstats.winsorize
----------------------------

Docstring has no 'Returns' section

scipy.stats.mstats.trimr
------------------------

Docstring has no 'Returns' section

scipy.stats.mstats.trimmed_std
------------------------------

Docstring has no 'Returns' section

scipy.stats.mstats.trimmed_std
------------------------------

scipy.stats.mstats.kendalltau_seasonal
--------------------------------------

Docstring has no 'Returns' section

scipy.stats.mstats.msign
------------------------

Docstring has no 'Returns' section

scipy.stats.mstats.trim
-----------------------

Docstring has no 'Returns' section

scipy.stats.mstats.rankdata
---------------------------

Docstring has no 'Returns' section

scipy.stats.mstats.obrientransform
----------------------------------

Docstring has no 'Returns' section

scipy.stats.mstats.trimmed_mean
-------------------------------

Docstring has no 'Returns' section

scipy.stats.mstats.scoreatpercentile
------------------------------------

Docstring has no 'Returns' section

scipy.stats.mstats.mjci
-----------------------

Docstring has no 'Returns' section

</details>

Of course, it is easily generalized to other sections, etc.

**Comment by WarrenWeckesser** [comment]
I've been using my own script [`find_docstring_issue.py`](https://github.com/WarrenWeckesser/analyze-scipy-code/blob/main/find_docstring_issues.py), but `numpydoc` probably already covers all that and more.  For example, here's the report generated by `numpydoc lint scipy/sparse/_construct.py:

<to be added...>

**Comment by ev-br** [comment]
We have some checking of docstrings in refguide-check.py still. Would be great to consolidate that with numpydoc validation and turn it on across the whole scipy codebase.
This a lot of churn though (SA02, PR08 above). Lots of "good first issue" material, too.

**Comment by lucascolley** [comment]
that would be a lot for the whole SciPy codebase 😅 but might be manageable if there is a good system for selecting which files to lint piece by piece. I'm using the `numpydoc` pre-commit hook in array-api-extra now (https://github.com/data-apis/array-api-extra/pull/67) and it's working well so far (modulo https://github.com/numpy/numpydoc/issues/561)

**Comment by Dronten** [comment]
is this issue still up ?

**Comment by lucascolley** [comment]
yes @Dronten ! Check out our contributor guide at https://scipy.github.io/devdocs/dev/index.html#scipy-development if you haven't already.

**Comment by lucascolley** [comment]
Sorry — I was wrong, it looks like this is already being tackled by gh-22527

**Comment by Dronten** [comment]
> Sorry — I was wrong, it looks like this is already being tackled by [gh-22527](https://github.com/scipy/scipy/pull/22527)

oh okay , no problem
